urllib3/util/connection.py

```diff
@@ -99,3 +106,39 @@ def _set_socket_options(sock, options):
 
     for opt in options:
         sock.setsockopt(*opt)
+
+
+def family_filter():
```

Comment:
I originally had `supported_ip_family()`, but I think `filter` works as well. Seems like it might be worth more discussion; I'll wait until there's a consensus on method name to re-commit.